types/field.go

```diff
@@ -37,3 +39,34 @@ func (l FieldsDefinition) Names() []string {
 	}
 	return names
 }
+
+// SelectedField is the public representation of a field selection during a graphql query
+type SelectedField struct {
+	Name             string
+	Alias            string
+	Directives       DirectiveList
+	Args             map[string]interface{}
+	Fields           []*SelectedField
+	AssertedTypeName string // non-empty for field selections on union fields
+}
+
+type SelectedFieldIdentifier struct {
+	Name  string
+	Alias string
+}
+
+// Lookup searches the field tree for the field defined by the input path
+func (f *SelectedField) Lookup(path ...SelectedFieldIdentifier) *SelectedField {
+	if len(path) == 0 || f == nil {
+		return f
+	}
+	if path[0].Name == "" {
+		panic("path component's name cannot be empty")
+	}
+	for _, subf := range f.Fields {
+		if subf.Name == path[0].Name && (path[0].Alias == "" || path[0].Alias == subf.Alias) {
```

Comment:
you need to ensure both match, i dont think it's an actually useful "feature" ("match first field by name") since it blocks a path where you'd have mixed aliased and non-aliased fields